### PR TITLE
SE-2 migration

### PR DESCRIPTION
**Pull request by carletex**
Have this ready before merging:

- [x] https://github.com/scaffold-eth/se-2-challenges/issues/138
- [ ] https://github.com/scaffold-eth/se-2-challenges/pull/136
- [x] Update server to use NodeJS >= v18.17

### CODE REVIEW

File: challenges.json [hunk 1]
```diff
@@ -2,37 +2,37 @@
...
   },
   {
     "id": 2,
     "name": "challenge-2-token-vendor",
-    "github": "https://github.com/scaffold-eth/scaffold-eth-challenges.git",
+    "github": "https://github.com/scaffold-eth/se-2-challenges.git",
     "contractName": "Vendor",
     "telegram": "https://t.me/joinchat/IfARhZFc5bfPwpjq",
-    "successMessage": "<p>You have successfully passed challenge 2! Keep it up with Challenge 3 - Dice Game!</p><p>--</p>"
+    "successMessage": "<p>You have successfully passed challenge 2!</p>p>You have passed the first three challenges on SpeedRunEthereum and can now join the BuidlGuidl! Head to your profile on SpeedRunEthereum.com and use the link to join. This will also unlock the Dice Game challenge and some more more challenging ones: Decentralized Exchange, Multisig, and SVG NFT challenges. These are more open ended which, once complete, can be submitted as a build on your buidlguidl.com portfolio to show off your web3 knowledge!</p><p>Join the next challenge’s Telegram channel on SpeedRunEthereum to get started. This channel includes other builders and BuidlGuild members to guide you along the way</p><p>--</p>"
```
Comment (rin-st) [inline challenges.json:22]: typo - `</p>p>`

**Comment by rin-st** [comment]
also noticed that in failedTest message described test file `(packages/hardhat/test/challenge_${challenge.id}.js)`, but in sre-2 files named like `ChallengeN.ts`

**Comment by carletex** [comment]
Thanks @rin-st, you are the man!

Fixed on https://github.com/austintgriffith/speedrun-grader/pull/9/commits/2d8cae3d6154da01afcca3afbc80e2b3d8cf7c2e

**Comment by technophile-04** [comment]
> (packages/hardhat/test/challenge_${challenge.id}.js)

Adding on to this, Since in new SE-2 we are calling State channel as no.5, when test fail we get this : 
Screenshot 2024-02-27 at 4 41 58 PM

Lol not a big deal, and we can ignore it since I think moving Challenge 9 => challenge 5 in autograder might require changes at multiple places like SRE webite etc and also might cause confusion in DB ? 

Incase we want to handle above wrong display of path in message we could something like thin in `message.js` : 
```diff
- {challenge.id}
+ {challenge.name.split("-")[1]}
```

Since we have control over `challenge` object (its a local json file controlled by us lol it might fine) 🙌

Also I went through the code only `message.js` needs to be updated rest everything displays properly great

**Comment by carletex** [comment]
Updated autograder server to node 18.17

**Comment by carletex** [comment]
I'm gonna merge this and deploy it live.

If everything goes well, I'll merge https://github.com/scaffold-eth/se-2-challenges/pull/136

**Pull request merged**